src/common/logging.cpp

```diff
@@ -28,34 +28,63 @@
 
 #include "logging.hpp"
 
+#include <assert.h>
 #include <stdarg.h>
-
+#include <stdint.h>
 #include <syslog.h>
 
-static int sLevel = LOG_INFO;
+static int        sLevel = LOG_INFO;
+static const char HEX_CHARS[] = "0123456789abcdef";
 
 void otbrLogInit(const char *aIdent, int aLevel)
 {
-    if (aLevel < LOG_EMERG || aLevel > LOG_DEBUG)
-    {
-        return;
-    }
+    assert(aIdent);
+    assert(aLevel >= LOG_EMERG && aLevel <= LOG_DEBUG);
+
     openlog(aIdent, LOG_CONS | LOG_PID | LOG_PERROR, LOG_USER);
     sLevel = aLevel;
 }
 
 void otbrLog(int aLevel, const char *aFormat, ...)
 {
-    if (aLevel < LOG_EMERG || aLevel > sLevel)
+    assert(aFormat);
+    assert(aLevel >= LOG_EMERG && aLevel <= LOG_DEBUG);
+
+    if (aLevel > sLevel)
     {
         return;
     }
+
     va_list args;
     va_start(args, aFormat);
     vsyslog(aLevel, aFormat, args);
     va_end(args);
 }
 
+void otbrDump(int aLevel, const char *aPrefix, const void *aMemory, size_t aSize)
```

Comment:
Thanks, I used the output of `otDump`. It provides more formatting. However, I didn't benefit much from the formatting.

1. Most dumping happens on binary data, it not that useful to provide ascii conversion.
2. When I check the dumps, I wanna simply copy and parse it with tools, those formatting is annoying because I have to concat those segments into the original hex string.

Thoughts?